src/components/Company/FederalTaxes/FederalTaxes.tsx

```diff
@@ -0,0 +1,137 @@
+import { ReactNode } from 'react'
+import { useFederalTaxDetailsUpdateMutation } from '@gusto/embedded-api/react-query/federalTaxDetailsUpdate'
+import { useFederalTaxDetailsGetSuspense } from '@gusto/embedded-api/react-query/federalTaxDetailsGet'
+import { FormProvider, useForm } from 'react-hook-form'
+import { Form as AriaForm } from 'react-aria-components'
+import { valibotResolver } from '@hookform/resolvers/valibot'
+import * as v from 'valibot'
+import {
+  TaxPayerType,
+  FilingForm,
+} from '@gusto/embedded-api/models/operations/putv1companiescompanyidfederaltaxdetails'
+import { FederalTaxDetails } from '@gusto/embedded-api/models/components/federaltaxdetails'
+import { Form } from '@/components/Company/FederalTaxes/Form'
+import { Actions } from '@/components/Company/FederalTaxes/Actions'
+import { Head } from '@/components/Company/FederalTaxes/Head'
+import { useI18n } from '@/i18n'
+import {
+  BaseComponent,
+  BaseComponentInterface,
+  CommonComponentInterface,
+  createCompoundContext,
+  useBase,
+} from '@/components/Base/Base'
+import { Flex } from '@/components/Common'
+import { companyEvents } from '@/shared/constants'
+
+export const FederalTaxFormSchema = v.object({
+  federalEin: v.optional(v.string()),
+  taxPayerType: v.optional(
+    v.union(Object.values(TaxPayerType).map(taxPayerType => v.literal(taxPayerType))),
+  ),
+  filingForm: v.optional(
+    v.union(Object.values(FilingForm).map(filingForm => v.literal(filingForm))),
+  ),
+  legalName: v.pipe(v.string(), v.nonEmpty()),
+})
+
+export type FederalTaxFormInputs = v.InferInput<typeof FederalTaxFormSchema>
+
+type FederalTaxesContextType = {
+  isPending: boolean
+  federalTaxDetails?: FederalTaxDetails
+}
+
+const [useFederalTaxes, FederalTaxesProvider] = createCompoundContext<FederalTaxesContextType>(
+  'CompanyFederalTaxesContext',
+)
+
+export { useFederalTaxes }
+
+interface FederalTaxesProps extends CommonComponentInterface {
+  companyId: string
+  children?: ReactNode
+  className?: string
+}
+
+export function FederalTaxes(props: FederalTaxesProps & BaseComponentInterface) {
+  return (
+    <BaseComponent {...props}>
+      <Root {...props}>{props.children}</Root>
+    </BaseComponent>
+  )
+}
+
+function Root({ companyId, children, className }: FederalTaxesProps) {
+  useI18n('Company.FederalTaxes')
+  const { onEvent, baseSubmitHandler } = useBase()
+
+  const {
+    data: { federalTaxDetails },
+  } = useFederalTaxDetailsGetSuspense({ companyId })
+  const { mutateAsync: updateFederalTaxDetails, isPending } = useFederalTaxDetailsUpdateMutation()
+
+  const formMethods = useForm<FederalTaxFormInputs>({
+    resolver: valibotResolver(FederalTaxFormSchema),
+    defaultValues: {
```

Comment:
Just curious, do we want to allow the consumer to provide defaultValues to override this ever? Looks like we're just putting the initial data in here which could be fine just doesn't sound like `defaultValues` :)

Reply:
Good callout! relevant thread just posted by @dmortal here in slack https://gustohq.slack.com/archives/C071WFJUNF5/p1741976496063599

I'm going to keep an eye on that and, based on the outcome, can ticket updating this form with default values as needed

Reply:
Sep ticket here to add default values https://gustohq.atlassian.net/browse/GWS-4214